nucypher/utilities/networking.py

```diff
@@ -136,12 +134,21 @@ def get_external_ip_from_default_teacher(network: str,
     Ursula.set_federated_mode(federated_only)
     #####
 
-    try:
-        teacher = Ursula.from_teacher_uri(teacher_uri=top_teacher_url,
-                                          federated_only=federated_only,
-                                          min_stake=0)  # TODO: Handle customized min stake here.
-    except NodeSeemsToBeDown:
-        # Teacher is unreachable.  Move on.
+    teacher = None
+    for teacher_uri in TEACHER_NODES[network]:
+        try:
+            teacher = Ursula.from_teacher_uri(teacher_uri=teacher_uri,
+                                              federated_only=federated_only,
+                                              min_stake=0)  # TODO: Handle customized min stake here.
+            # Found a reachable teacher, return from loop
+            break
```

Comment:
Do we need to also incorporate L155 as a check in this loop? That being said if `from_teacher_uri` works (it makes a connection to the node) chances are `_request_from_node` should work anyway (it also makes a connection to the node) - so it may be superfluous. Interested in what others think (@KPrasch )?

Reply:
I think it's a good idea because `_request_from_node` is handling parsing and related errors.